### PR TITLE
LPS-169559 Add LIst Type Entries ERC on Local Service and Rest Layers

### PR DESCRIPTION
**Pull request by mateussgn**
Related issue: https://issues.liferay.com/browse/LPS-169559

### CODE REVIEW

**Comment by mateussgn** [comment]
ci:test:sf

File: modules/apps/list-type/list-type-service/service.xml [hunk 1]
```diff
@@ -55,13 +55,18 @@
 		<finder name="ListTypeDefinitionId" return-type="Collection">
 			<finder-column name="listTypeDefinitionId" />
 		</finder>
+		<finder name="ERC_C" return-type="ListTypeEntry">
+			<finder-column name="externalReferenceCode"></finder-column>
+			<finder-column name="companyId"></finder-column>
```
Comment (marcelabc) [inline modules/apps/list-type/list-type-service/service.xml:60]: Why are you filtering by company and not for the "listTypeDefinitionId"?

File: modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryLocalServiceImpl.java [hunk 2]
```diff
@@ -53,14 +54,16 @@ public ListTypeEntry addListTypeEntry(
 			long listTypeDefinitionId, String key, Map<Locale, String> nameMap)
 		throws PortalException {
 
+		User user = _userLocalService.getUser(userId);
+
 		_validateKey(listTypeDefinitionId, key);
+		_validateExternalReferenceCode(
```
Comment (marcelabc) [inline modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryLocalServiceImpl.java:62]: Could you sort the method calls, please?

File: modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryLocalServiceImpl.java [hunk 3]
```diff
@@ -135,6 +142,21 @@ public ListTypeEntry updateListTypeEntry(
 		return listTypeEntryPersistence.update(listTypeEntry);
 	}
 
+	private void _validateExternalReferenceCode(
+			String externalReferenceCode, long companyId, long listTypeEntryId)
```
Comment (marcelabc) [inline modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryLocalServiceImpl.java:156]: I think that we should validate by "listTypeDefinitionId" and not by company. What do you think?

File: modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryLocalServiceImpl.java [hunk 4]
```diff
@@ -129,9 +131,17 @@ public ListTypeEntry updateListTypeEntry(
...
+			_validateExternalReferenceCode(
+				externalReferenceCode, listTypeEntry.getCompanyId(),
+				listTypeEntryId);
+		}
+
+		if (!GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-169559")) &&
+			Validator.isNull(externalReferenceCode)) {
+
+			externalReferenceCode = listTypeEntry.getKey();
+		}
```
Comment (marcelabc) [inline modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryLocalServiceImpl.java:144]: I don't believe we need this condition, I think we can simplify this code:

```
if (GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-169559")) && Validator.isNotNull(externalReferenceCode)) {
	_validateExternalReferenceCode();

       listTypeEntry.setExternalReferenceCode(externalReferenceCode);
}
```

File: modules/apps/headless/headless-admin-list-type/headless-admin-list-type-impl/src/main/java/com/liferay/headless/admin/list/type/internal/dto/v1_0/util/ListTypeEntryUtil.java [hunk 5]
```diff
@@ -33,6 +33,8 @@ public static ListTypeEntry toListTypeEntry(
 			{
 				dateCreated = serviceBuilderListTypeEntry.getCreateDate();
 				dateModified = serviceBuilderListTypeEntry.getModifiedDate();
+				externalReferenceCode =
```
Comment (marcelabc) [inline modules/apps/headless/headless-admin-list-type/headless-admin-list-type-impl/src/main/java/com/liferay/headless/admin/list/type/internal/dto/v1_0/util/ListTypeEntryUtil.java:36]: We need to put the Feature Flag here :+1:

File: modules/apps/list-type/list-type-service/service.xml (hunk 1 above)
Comment (marcelabc) [reply modules/apps/list-type/list-type-service/service.xml:60]: I think we need to scope by companyId + listTypeDefinitionId. 
(The object field is scoped like this).

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:stop

**Comment by mateussgn** [comment]
ci:close

**Comment by mateussgn** [comment]
Resent: https://github.com/liferay-objects/liferay-portal/pull/1508

**Pull request closed without merging**